src/lib/gw2/itemStats.js

```diff
@@ -0,0 +1,90 @@
+  );
+}
+
+function getCalcFunctionStandard (itemAttributes, selectedStat) {
+  const majorAttributeKeys = getMajorKeys(itemAttributes, selectedStat);
+  const calcMajor = (v) => calcAttribute(parseInt(itemAttributes.majorAttribute, 10), 32, v);
+  const calcMinor = (v) => calcAttribute(parseInt(itemAttributes.minorAttribute, 10), 18, v);
+  const isMajor = (attr) => (majorAttributeKeys.indexOf(attr) > -1);
+  return (attr, v) => (isMajor(attr) ? calcMajor(v) : calcMinor(v));
+}
+
+function getCalcFunctionQuad (itemAttributes, selectedStat) {
+  const majorAttributeKeys = getMajorKeys(itemAttributes, selectedStat);
+  const calcMajor = (v) => calcAttribute(parseInt(itemAttributes.majorQuadAttribute, 10), 25, v);
+  const calcMinor = (v) => calcAttribute(parseInt(itemAttributes.minorQuadAttribute, 10), 12, v);
+  const isMajor = (attr) => (majorAttributeKeys.indexOf(attr) > -1);
+  return (attr, v) => (isMajor(attr) ? calcMajor(v) : calcMinor(v));
+}
+
+function getCalcFunctionQuadOld (itemAttributes, selectedStat) {
+  const minorValue = parseInt(itemAttributes.minorAttribute, 10);
+  const majorAttributeKeys = getMajorKeys(itemAttributes, selectedStat);
+  const calcMajor = (v) => calcAttribute(parseInt(itemAttributes.majorAttribute, 10), 32, v);
+  const calcMinor = (v) => calcAttribute(minorValue, 18, v);
+
+  // This is the most hacky thing ever, but we have 2 items that we need
+  // to detect by the position in attributes.
+  const isMajorMinor = (attr) => Object.keys(selectedStat.attributes).indexOf(attr) === 3;
+  const isMajor = (attr) => (majorAttributeKeys.indexOf(attr) > -1);
+  return (attr, v) => {
+    if (isMajor(attr)) {
+      return calcMajor(v);
+    }
+    if (isMajorMinor(attr)) {
+      return calcMinor(v);
+    }
+
+    if (v === 0) {
+      return calcAttribute(0, 18, v);
+    }
+    return calcAttribute(minorValue, 0, v);
+  };
+}
+
+function getCalcFunctionCelectiel (itemAttributes) {
+  return () => calcAttribute(parseInt(itemAttributes.celestialNbr, 10), 13);
+}
+
+function getCalcFunction (itemAttributes, selectedStat) {
+  if (selectedStat.name.includes(' and ')) {
+    return getCalcFunctionQuadOld(itemAttributes, selectedStat);
+  }
+
+  switch (Object.keys(selectedStat.attributes).length) {
+    case 3:
+      return getCalcFunctionStandard(itemAttributes, selectedStat);
+    case 4:
+      return getCalcFunctionQuad(itemAttributes, selectedStat);
+    case 7:
+      return getCalcFunctionCelectiel(itemAttributes);
+    default:
+      throw new Error('impossible condition');
+  }
+}
+
+export default function applyAttributes (item, selectedStat) {
+  const itemAttributes = queryItemAttributes(item.details.type.toLowerCase(),
+    item.rarity.toLowerCase(),
+    item.level);
+  const calcModifier = getCalcFunction(itemAttributes, selectedStat);
+  return Object.keys(selectedStat.attributes)
+    .map((attribute) => ({
+      attribute,
+      // TODO: need to calculate modifier, how?
```

Comment:
I'm meaning here that at the moment the modifier (on item) needs to be calculated using the itemstat object from the server.
the thing is, there is no sane formula i was able to find, so i'm counting on itemstats object for which stats should be applied, but apply using internal calculation from gw2wiki which is not a real calculation as the game would do it.